clients/admin-ui/src/features/system/SystemInformationForm.tsx

```diff
@@ -219,7 +253,10 @@ const SystemInformationForm = ({
 
             <SystemFormInputGroup heading="System details">
               {features.dictionaryService ? (
-                <VendorSelector options={dictionaryOptions} />
+                <VendorSelector
+                  options={dictionaryOptions}
+                  handleVendorSelected={handleVendorSelected}
```

Comment:
this is a nit, and it's only something I'm just starting to form an opinion on, but I think this should be `onVendorSelected`. the idea being, _when_ a vendor is selected, we execute the function passed in. `handleVendorSelected` feels more like "do the actual handling of vendor selection" (i.e. setting it in the form, or something)